File: mobile/app/src/main/java/ua/nure/cleaningservice/ui/rva/SearchRVA.java
package ua.nure.cleaningservice.ui.rva;

import android.content.Context;
import android.content.Intent;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.cardview.widget.CardView;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

import ua.nure.cleaningservice.R;
import ua.nure.cleaningservice.data.Company;
import ua.nure.cleaningservice.ui.contracts.SignContractActivity;

public class SearchRVA extends RecyclerView.Adapter<SearchRVA.SearchViewHolder>{

    private Context mContext;

    public SearchViewHolder onCreateViewHolder(ViewGroup viewGroup, int i) {
        View v = LayoutInflater.from(viewGroup.getContext()).inflate(R.layout.search_card, viewGroup, false);
        SearchViewHolder searchViewHolder = new SearchViewHolder(v);
        return searchViewHolder;
    }

    private List<Company> mCompanies;

    public SearchRVA(Context context, List<Company> companies){
        this.mContext = context;
        this.mCompanies = companies;
    }

    @Override
    public void onBindViewHolder(@NonNull SearchViewHolder holder, int position) {
        holder.mCardView.setId(Integer.parseInt(mCompanies.get(position).getId()));
        holder.mNameTV.setText(mCompanies.get(position).getName());
        holder.mPhoneTV.setText(mCompanies.get(position).getPhoneNumber());
        holder.mEmailTV.setText(mCompanies.get(position).getEmail());
        holder.mSignButton.setOnClickListener(v -> {
            signContract(holder);
        });
    }
    @Override
    public void onAttachedToRecyclerView(RecyclerView recyclerView) {
        super.onAttachedToRecyclerView(recyclerView);
    }

    @Override
    public int getItemCount() {
        return mCompanies.size();
    }

    class SearchViewHolder extends RecyclerView.ViewHolder {
        CardView mCardView;
        TextView mNameTV, mPhoneTV, mEmailTV;
        Button mSignButton;
        SearchViewHolder(View itemView) {
            super(itemView);
            mCardView = (CardView) itemView.findViewById(R.id.search_cv);
            mNameTV = (TextView) itemView.findViewById(R.id.name_search_label);
            mPhoneTV = (TextView) itemView.findViewById(R.id.phone_text);
            mEmailTV = (TextView) itemView.findViewById(R.id.email_text);
            mSignButton = (Button) itemView.findViewById(R.id.sign_contract_btn);
        }

    }

    private void signContract(SearchViewHolder holder) {
        Intent intent = new Intent(mContext, SignContractActivity.class);
        intent.putExtra("cEmail", holder.mEmailTV.getText());
        mContext.startActivity(intent);
    }


}

File: mobile/app/src/main/java/ua/nure/cleaningservice/ui/rva/SearchRVA.kt
package ua.nure.cleaningservice.ui.rva

import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import android.widget.TextView
import androidx.cardview.widget.CardView
import androidx.recyclerview.widget.RecyclerView
import ua.nure.cleaningservice.R
import ua.nure.cleaningservice.data.User
import ua.nure.cleaningservice.ui.contracts.SignContractActivity
import ua.nure.cleaningservice.ui.rva.SearchRVA.SearchViewHolder

class SearchRVA(private val mContext: Context, private val mUsers: List<User>?) : RecyclerView.Adapter<SearchViewHolder>() {
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): SearchViewHolder {
        val v = LayoutInflater.from(viewGroup.context).inflate(R.layout.search_card, viewGroup, false)
        return SearchViewHolder(v)
    }

    override fun onBindViewHolder(holder: SearchViewHolder, position: Int) {
        holder.mCardView.id = mUsers!![position].id.toInt()
        holder.mNameTV.text = mUsers[position].name
        holder.mPhoneTV.text = mUsers[position].phoneNumber
        holder.mEmailTV.text = mUsers[position].email
        holder.mSignButton.setOnClickListener { v: View? -> signContract(holder) }
    }

    override fun onAttachedToRecyclerView(recyclerView: RecyclerView) {
        super.onAttachedToRecyclerView(recyclerView)
    }

    override fun getItemCount(): Int {
        return mUsers!!.size
    }

    inner class SearchViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var mCardView: CardView
        var mNameTV: TextView
        var mPhoneTV: TextView
        var mEmailTV: TextView
        var mSignButton: Button

        init {
            mCardView = itemView.findViewById<View>(R.id.search_cv) as CardView
            mNameTV = itemView.findViewById<View>(R.id.name_search_label) as TextView
            mPhoneTV = itemView.findViewById<View>(R.id.phone_text) as TextView
            mEmailTV = itemView.findViewById<View>(R.id.email_text) as TextView
            mSignButton = itemView.findViewById<View>(R.id.sign_contract_btn) as Button
        }
    }

    private fun signContract(holder: SearchViewHolder) {
        val intent = Intent(mContext, SignContractActivity::class.java)
        intent.putExtra("cEmail", holder.mEmailTV.text)
        mContext.startActivity(intent)
    }
}
